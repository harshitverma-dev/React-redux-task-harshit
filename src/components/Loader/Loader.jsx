import React from 'react'

const Loader = () => {
    return (
        <div className='loader-container' data-testid="loader-container">
            <svg
                className="loader"
                width="40"
                height="40"
                viewBox="0 0 40 40"
                xmlns="http://www.w3.org/2000/svg"
                stroke="#4F46E5"
                data-testid="loader-svg"
            >
                <g fill="none" fillRule="evenodd">
                    <g transform="translate(2 2)" strokeWidth="3">
                        <circle strokeOpacity=".2" cx="18" cy="18" r="18" />
                        <path d="M36 18c0-9.94-8.06-18-18-18">
                            <animateTransform
                                attributeName="transform"
                                type="rotate"
                                from="0 18 18"
                                to="360 18 18"
                                dur=".8s"
                                repeatCount="indefinite"
                            />
                        </path>
                    </g>
                </g>
            </svg>
        </div>

    )
}

export default Loader