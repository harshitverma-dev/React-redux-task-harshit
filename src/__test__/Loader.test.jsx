import { describe, it, expect } from 'vitest';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Loader from '../components/Loader/Loader';
import React from 'react';

describe('Loader Component', () => {
    it('renders loader container', () => {
        render(<Loader />);
        const loaderContainer = screen.getByTestId('loader-container');
        expect(loaderContainer).toBeInTheDocument();
    });

    it('renders SVG element', () => {
        render(<Loader />);
        const svgElement = screen.getByTestId('loader-svg');
        expect(svgElement).toBeInTheDocument();
        expect(svgElement).toHaveAttribute('width', '40');
        expect(svgElement).toHaveAttribute('height', '40');
    });

    it('has correct stroke color', () => {
        render(<Loader />);
        const svgElement = screen.getByTestId('loader-svg');
        expect(svgElement).toHaveAttribute('stroke', '#4F46E5');
    });

    it('matches snapshot', () => {
        const { container } = render(<Loader />);
        expect(container).toMatchSnapshot();
    });
});