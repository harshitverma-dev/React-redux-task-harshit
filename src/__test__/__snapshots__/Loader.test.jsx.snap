// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Loader Component > matches snapshot 1`] = `
<div>
  <div
    class="loader-container"
    data-testid="loader-container"
  >
    <svg
      class="loader"
      data-testid="loader-svg"
      height="40"
      stroke="#4F46E5"
      viewBox="0 0 40 40"
      width="40"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        fill="none"
        fill-rule="evenodd"
      >
        <g
          stroke-width="3"
          transform="translate(2 2)"
        >
          <circle
            cx="18"
            cy="18"
            r="18"
            stroke-opacity=".2"
          />
          <path
            d="M36 18c0-9.94-8.06-18-18-18"
          >
            <animatetransform
              attributeName="transform"
              dur=".8s"
              from="0 18 18"
              repeatCount="indefinite"
              to="360 18 18"
              type="rotate"
            />
          </path>
        </g>
      </g>
    </svg>
  </div>
</div>
`;
